source/common/grpc/google_grpc_utils.cc

```diff
@@ -0,0 +1,111 @@
+#include "common/grpc/google_grpc_utils.h"
+
+#include <atomic>
+#include <cstdint>
+#include <cstring>
+#include <string>
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/assert.h"
+#include "common/common/empty_string.h"
+#include "common/common/enum_to_int.h"
+#include "common/common/fmt.h"
+#include "common/common/macros.h"
+#include "common/common/stack_array.h"
+#include "common/common/utility.h"
+
+#include "absl/strings/match.h"
+
+namespace Envoy {
+namespace Grpc {
+
+struct BufferInstanceContainer {
+  BufferInstanceContainer(int ref_count, Buffer::InstancePtr&& buffer)
+      : ref_count_(ref_count), buffer_(std::move(buffer)) {}
+  std::atomic<uint32_t> ref_count_; // In case gPRC dereferences in a different threads.
+  Buffer::InstancePtr buffer_;
+
+  static void derefBufferInstanceContainer(void* container_ptr) {
+    auto container = reinterpret_cast<BufferInstanceContainer*>(container_ptr);
+    container->ref_count_--;
+    // This is safe because the ref_count_ is never incremented.
+    if (container->ref_count_ <= 0) {
+      delete container;
+    }
+  }
+};
+
+grpc::ByteBuffer GoogleGrpcUtils::makeByteBuffer(Buffer::InstancePtr&& buffer_instance) {
+  if (!buffer_instance) {
+    return {};
+  }
+  Buffer::RawSlice on_raw_slice;
+  // NB: we need to pass in >= 1 in order to get the real "n" (see Buffer::Instance for details).
+  int n_slices = buffer_instance->getRawSlices(&on_raw_slice, 1);
```

Comment:
Nit: generally prefer `const` for immutable temps.

Reply:
Done

Reply:
Still not `const` :)

Reply:
done